### PR TITLE
feat: adds new edition options

### PR DESCRIPTION
**Pull request by kendallstrautman**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-ksadds-new-tfc-edition-options-hashicorp.vercel.app/terraform/tutorials/cloud/cloud-agents) 🔎
- [Asana task](https://app.asana.com/0/1199634971449915/1203675527903646) 🎟️

## 🗒️ What

This PR adds two new edition types to align with Terraforms upcoming pricing changes. The legacy types are there so we can ease the migration as authors go to update all source content with the new editions. 

[This follow up task](https://app.asana.com/0/1203652232454030/1203704507579626) involves removing the old options once the content is updated by March. 

See related PRs:
- https://github.com/hashicorp/learn-api-content-sync/pull/67
- https://github.com/hashicorp/learn-api/pull/146

## To test

There is no way to test on the preview since the content doesn't use these editions. I seeded a local api / database with the updates in those backend PRs, and tested the frontmatter with the new edition types. Here is a picture of the new options in use (see badges list in the tutorial meta): 
 
Screen Shot 2023-01-11 at 4 08 48 PM

Screen Shot 2023-01-11 at 4 07 45 PM

However, all the other edition types should still work. And there should be no changes to[ the tutorial view.](https://dev-portal-git-ksadds-new-tfc-edition-options-hashicorp.vercel.app/terraform/tutorials/cloud/cloud-agents)

### CODE REVIEW

File: src/components/tutorial-meta/components/badges/helpers.tsx [hunk 1]
```diff
@@ -5,13 +5,20 @@ import Badge from 'components/badge'
 import ProductIcon from 'components/product-icon'
 
 const editionDisplayOptions: { [K in EditionOption]: string } = {
+	[EditionOption.enterprise]: 'Enterprise',
+	[EditionOption.hcp]: 'HCP',
+	[EditionOption.openSource]: 'Open Source',
+	[EditionOption.tfcStandard]: 'Standard',
```
Comment (kendallstrautman) [inline src/components/tutorial-meta/components/badges/helpers.tsx:11]:
```suggestion
	[EditionOption.openSource]: 'Open Source',
	[EditionOption.hcp]: 'HCP',
	[EditionOption.tfcStandard]: 'Standard',
```

File: src/components/tutorial-meta/components/badges/helpers.tsx [hunk 2]
```diff
@@ -5,13 +5,20 @@ import Badge from 'components/badge'
 import ProductIcon from 'components/product-icon'
 
 const editionDisplayOptions: { [K in EditionOption]: string } = {
+	[EditionOption.enterprise]: 'Enterprise',
+	[EditionOption.openSource]: 'Open Source',
```
Comment (kendallstrautman) [inline src/components/tutorial-meta/components/badges/helpers.tsx:9]:
```suggestion
	[EditionOption.openSource]: 'Open Source',
	[EditionOption.enterprise]: 'Enterprise',
```

Comment (kendallstrautman) [reply src/components/tutorial-meta/components/badges/helpers.tsx]: Just cleaning up the order

**Review by kylemac — APPROVED** [review]
approved, given the updated code is legit and the screenshots verify 👍

**Pull request merged**